backend/lib/piece-indexer.js

```diff
@@ -0,0 +1,82 @@
+import { fetchPayloadCid } from './pix-service/service.js'
+import { parseDeals } from './deal-observer.js'
+import assert from 'node:assert'
+import * as util from 'node:util'
+
+/** @import {Queryable} from '@filecoin-station/deal-observer-db' */
+/** @import { Static } from '@sinclair/typebox' */
+/** @import { ActiveDealDbEntry } from '@filecoin-station/deal-observer-db/lib/types.js' */
+
+/**
+ * @param {Queryable} pgPool
+ * @param {function} makeRpcRequest
+ * @param {function} makePixRequest
+ * @returns {Promise<void>}
+ */
+export async function updatePayloadCids (pgPool, makeRpcRequest, activeDeals, makePixRequest) {
+  const updatedDeals = []
+  for (const deal of activeDeals) {
+    const payloadCid = await fetchPayloadCid(deal.miner_id, deal.piece_cid, makeRpcRequest, makePixRequest)
+    deal.payload_cid = payloadCid
+    updatedDeals.push(deal)
+  }
+  await updatePayloadInActiveDeal(pgPool, updatedDeals)
```

Comment:
Since this is updating multiple deals at once, the name should be plural.

```suggestion
  await updatePayloadInActiveDeals(pgPool, updatedDeals)
```